src/routes/reservations/reservation.svelte

```diff
@@ -0,0 +1,204 @@
+<script>
+    // Framework imports
+    import { onMount } from "svelte";
+    import { urls } from "$lib/util.js";
+    import { popup } from "@skeletonlabs/skeleton";
+    import { getToastStore } from "@skeletonlabs/skeleton";
+    // Import 🐦
+    import { ExponentialBackoff, handleAll, retry } from "cockatiel";
+
+    // Definitions
+
+    // Export data from parent component
+    export let reservationData;
+    export let cancellable;
+
+    // Constants
+    const toastStore = getToastStore();
+
+    const options = {
+        year: "numeric",
+        month: "numeric",
+        day: "numeric",
+        hour: "numeric",
+        minute: "numeric",
+        second: "numeric",
+        hour12: false,
+    };
+
+    // Toast Settings
+    const toastSuccsess = {
+        message: "Reservierung wurde erfolgreich gelöscht",
+        hideDismiss: true, // Hide the dismiss button on toast
+        timeout: 3000, // Auto dismiss toast after 3 seconds
+        background: "variant-filled-secondary",
+    };
+
+    const toastError = {
+        message: "Reservierung konnte nicht gelöscht werden",
+        hideDismiss: true, // Hide the dismiss button on toast
+        timeout: 3000, // Auto dismiss toast after 3 seconds
+        background: "variant-filled-error",
+    };
+
+    // Policy for restarting backend fetched up to 5 times if there's no reply
+    const retryPolicy = retry(handleAll, {
+        maxAttempts: 5,
+        backoff: new ExponentialBackoff(),
+    });
+
+    // Popup settings
+    const popupClick = {
+        event: "click",
+        target: "popupClick",
+        placement: "top",
+    };
+
+    let carData = new Promise((resolve, reject) => {});
```

Comment:
Warum das? Ein Promise, der nur ein leeres Objekt zurückgibt?

Reply:
Initialisierung auf ein leeren promise, damit keine Fehler sofort geworfen werden, wenn das undefined ist

Reply:
Hmm okay

Reply:
Ist in der Tat nicht super clean, aber ich wusste aktuell auch keine bessere Lösung